main_test.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestVerifySignature(t *testing.T) {
+	// Using a scorecard results file from a previous run that was successfully signed.
+	// This isn't expected to pass verifyScorecardWorkflow because the workflow it was
+	// generated from contains extra steps to call cosign.
+	sarifpayload, _ := ioutil.ReadFile("testdata/validSig-invalidWkflw.sarif")
+	jsonpayload, _ := ioutil.ReadFile("testdata/scorecard-results.json")
+	payload := ScorecardOutput{SarifOutput: string(sarifpayload), JsonOutput: string(jsonpayload)}
+	payloadbytes, err := json.Marshal(payload)
+	assert.Equal(t, err, nil)
+
+	r, _ := http.NewRequest("POST", "/verify", bytes.NewBuffer(payloadbytes))
+	r.Header = http.Header{"Repository": []string{"rohankh532/scorecard-OIDC-test"}, "Branch": []string{"refs/heads/main"}}
```

Comment:
should non standard http headers be prepended with `X-`?

Reply:
yeah, fixed